config/provider.go

```diff
@@ -25,6 +25,9 @@ import "fmt"
 // ConfigurationChangeCallback is called for updates of configuration data
 type ConfigurationChangeCallback func(key string, provider string, configdata interface{})
 
+// Root marks the root node in a Provider
+const Root = ""
```

Comment:
You mean actual types or or just const strings? I am not aware how can we do the former..